server/src/main/java/io/crate/protocols/http/Headers.java

```diff
@@ -63,15 +63,33 @@ public static void setKeepAlive(HttpVersion httpVersion, FullHttpResponse resp)
         }
     }
 
-    public static Credentials extractCredentialsFromHttpBasicAuthHeader(String authHeaderValue) {
+    /**
+     * An entry point for HTTP authentication, forwards to either Basic or JWT, depending on header.
+     * @param authHeaderValue contains authentication schema (basic or bearer) and auth payload (token or password) separated by space.
+     * Authentication schema is case-insensitive: <a href="https://datatracker.ietf.org/doc/html/rfc7235?ref=blog.teknkl.com#section-2.1">spec</a>
+     */
+    public static Credentials extractCredentialsFromHttpAuthHeader(String authHeaderValue) {
         if (authHeaderValue == null || authHeaderValue.isEmpty()) {
             // Empty credentials.
             return new Credentials("", new char[] {});
         }
+        if (authHeaderValue.substring(0, 5).equalsIgnoreCase("basic")) {
+            // Account to space
+            return extractCredentialsFromHttpBasicAuthHeader(authHeaderValue.substring(6));
+        } else {
+            if (authHeaderValue.substring(0, 6).equalsIgnoreCase("bearer")) {
+                // Account to space
+                return new Credentials(authHeaderValue.substring(7));
+            } else {
+                throw new IllegalArgumentException("Only basic or bearer HTTP Authentication schemas are allowed.");
+            }
+        }
```

Comment:
As, according to the standard, a token is not allowed to contain a whitespace character, this could maybe simplified by using `authHeaderValue.split(" ")`, followed by a switch stmt on the first value(auth schema).

Reply:
afaik username or password can contain whitespace in Basic auth.

From https://www.rfc-editor.org/rfc/rfc7617.html

>  If the user agent wishes to send the user-id "Aladdin" and password
   "open sesame", it would use the following header field:

      Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==
      
Also found https://github.com/Huachao/vscode-restclient/issues/896 where whitespace in name is a problem.

If we do `split`, we can resolve schema with `switch` but then for `Basic` we would need either concat split values again or use `substring` anyway.

Reply:
Isn't the `credentials` (or also sometimes referred as `token`) always expected to be Base64 encoded like in the example you provided? Also defined here like that: https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes

We currently expect it do be Base64 encoded as well https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/protocols/http/Headers.java#L76.

Reply:
Yes, forgot about base64. 
Simplified in https://github.com/crate/crate/commit/2f59002b2f84f11deac19b2bcd83e4a86f4de563